numba/core/types/containers.py

```diff
@@ -433,6 +434,34 @@ def __getitem__(self, args):
         return self.dtype
 
 
+class LiteralList(Literal, _HeterogeneousTuple):
+
+    mutable = False
+
+    def __init__(self, literal_value):
+        _HeterogeneousTuple.is_types_iterable(literal_value)
+        self._literal_init(list(literal_value))
+        tys = [unliteral(x) for x in literal_value]
+        self.types = tuple(literal_value)
+        self.count = len(self.types)
+        self.name = 'LiteralList({})'.format(literal_value)
+        literal_vals = [getattr(x, 'literal_value', None) for x in literal_value]
+
+    def __unliteral__(self):
+        return self
```

Comment:
Couldn't this degrade into a list with initial_value?

Reply:
Possibly, but only if the types are coercible? e.g. `['a', 1, 'b', 2]` couldn't degrade.

Reply:
hmmmm, I'm wondering now how this is reachable, if the types are coercible on construction, a `List` with an `initial_value` will be constructed, if they are not a `LiteralList` is constructed. If the latter, there's no way it can degrade to a `List` as it's not type coercible in the first place. The only situation I think this could occur in is if something is deliberately spoofing a signature for some purpose.

Reply:
Resolved by Poison type, it's essentially invalid.